SpiralMatrix.java

```diff
@@ -0,0 +1,65 @@
+// Time Complexity : O(N * M) N = rows, M = columns
+// Space Complexity : O(1)
+// Did this code successfully run on Leetcode : Yes
+// Three line explanation of solution in plain english
+//1. Maintain four pointers Top, Bottom, Left, Right to traverse though the matrix.
```

Comment:
A good example for this would be- 
-keep a track of four pointers representing each boundary, after traversing and adding
-to array one of 4 side which constitute a spiral traversal, shrink the boundary.
-repeat the same until the boundaries haven't crossed each other.